dynamic-proxy/src/main/java/com/iluwatar/dynamicproxy/tinyrestclient/TinyRestClient.java

```diff
@@ -0,0 +1,157 @@
+ * It is supported by the HttpClient Java library.
+ */
+public class TinyRestClient {
+
+  private String baseUrl;
+  private HttpClient httpClient;
+
+  /**
+   * Class constructor.
+   *
+   * @param baseUrl    Root url for endpoints.
+   * @param httpClient Handle the http communication.
+   */
+  public TinyRestClient(String baseUrl, HttpClient httpClient) {
+    this.baseUrl = baseUrl;
+    this.httpClient = httpClient;
+  }
+
+  /**
+   * Creates a http communication to request and receive data from an endpoint.
+   *
+   * @param method Interface's method which is annotated with a http method.
+   * @param args   Method's arguments passed in the call.
+   * @return Response from the endpoint.
+   * @throws IOException          Exception thrown when any fail happens in the call.
+   * @throws InterruptedException Exception thrown when call is interrupted.
+   */
+  public Object send(Method method, Object[] args) throws IOException, InterruptedException {
+    var httpMethodAnnotation = getHttpMethodAnnotation(method.getDeclaredAnnotations());
+    if (httpMethodAnnotation == null) {
+      return  null;
+    }
+    var httpMethodName = httpMethodAnnotation.annotationType().getSimpleName().toUpperCase();
+    var url = baseUrl + buildUrl(method, args, httpMethodAnnotation);
+    var bodyPublisher = buildBodyPublisher(method, args);
+    var httpRequest = HttpRequest.newBuilder()
+        .uri(URI.create(url))
+        .header("Content-Type", "application/json")
+        .method(httpMethodName, bodyPublisher)
+        .build();
+    var httpResponse = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+    return getResponse(method, httpResponse);
+  }
+
+  private String buildUrl(Method method, Object[] args, Annotation httpMethodAnnotation) {
+    var url = annotationValue(httpMethodAnnotation);
+    if (url == null) {
+      return "";
+    }
+    var index = 0;
+    for (var parameter : method.getParameters()) {
+      var pathAnnotation = getAnnotationOf(parameter.getDeclaredAnnotations(), Path.class);
+      if (pathAnnotation != null) {
+        var pathParam = "{" + annotationValue(pathAnnotation) + "}";
+        var pathValue = args[index].toString();
+        url = url.replace(pathParam, pathValue);
+      }
+      index++;
+    }
+    return url;
+  }
+
+  private HttpRequest.BodyPublisher buildBodyPublisher(Method method, Object[] args) {
+    var index = 0;
+    for (var parameter : method.getParameters()) {
+      var bodyAnnotation = getAnnotationOf(parameter.getDeclaredAnnotations(), Body.class);
+      if (bodyAnnotation != null) {
+        var body = JsonUtil.objectToJson(args[index]);
+        return HttpRequest.BodyPublishers.ofString(body);
+      }
+      index++;
+    }
+    return HttpRequest.BodyPublishers.noBody();
+  }
+
+  private Object getResponse(Method method, HttpResponse<String> httpResponse) {
+    var rawData = httpResponse.body();
+    var returnType = method.getGenericReturnType();
```

Comment:
Consider adding checks for safer casting.

Reply:
Done. A check was added.